deployment/build-and-stage.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO(michaelkedar): This is intentionally separate from staging.yaml
+# to not break existing deployment process as new one is implemented
+# eventually, they can probably be combined
+
+steps:
+# TODO(michaelkedar): Docker image creation & tagging
+
+# Apply Terraform config to staging environment
+# TODO(michaelkedar): regenerate the api protobufs?
+- name: gcr.io/oss-vdb/terraform
+  args: ['init', '-no-color']
+  dir:  deployment/terraform/environments/oss-vdb-test
+- name: gcr.io/oss-vdb/terraform
+  args: ['apply', '-no-color', '-auto-approve']
+  dir:  deployment/terraform/environments/oss-vdb-test
+
+# TODO(michaelkedar): App Engine for oss-vdb-test
+# TODO(michaelkedar): Cloud Deploy
+
+timeout: 7200s
+# Set in Cloud Build:
+# serviceAccount: 'projects/oss-vdb/serviceAccounts/deployment@oss-vdb.iam.gserviceaccount.com'
+
+# TODO(michaelkedar): the terraform bucket is not the best place for this
```

Comment:
This bucket seems reasonable to me actually. Only comment would be to rename the "directory" "apply-logs"

Reply:
It's just that eventually this will also be building the container images and deploying to App Engine / Cloud Deploy, which I didn't really think belonged in the terraform bucket.

Reply:
I think it's okay to have reasonably well-affiliated things in the one bucket, rather than having a proliferation of buckets